routes/api.js

```diff
@@ -1,10 +1,74 @@
 const express = require('express');
+const jwt    = require('jsonwebtoken');
+const config = require('../config/settings');
 const router = express.Router();
+const models = require('../database/models');
+
+const localUser = models.user_account;
+const progress = models.progress;
+
+router.use(function(req, res, next) {
+
+    // check header or url parameters or post parameters for token
+    let token = req.body.token || req.query.token || req.headers['x-access-token'];
```

Comment:
you have both `body` & `query` statement in here, one is for `post` while the other is for `get`.
so, what happens in a get query, where body might be undefined?
you can't get `token` of undefined.